src/renderers/dom/client/utils/setInnerHTML.js

```diff
@@ -28,7 +32,19 @@ var createMicrosoftUnsafeLocalFunction = require('createMicrosoftUnsafeLocalFunc
  */
 var setInnerHTML = createMicrosoftUnsafeLocalFunction(
   function(node, html) {
-    node.innerHTML = html;
+    // IE does not have innerHTML for SVG nodes, so instead we inject the
+    // new markup in a temp node and then move the child nodes across into
+    // the target node
+    if (node.namespaceURI === DOMNamespaces.svg && !('innerHTML' in node)) {
+      reusableSVGContainer = reusableSVGContainer || document.createElement('div');
+      reusableSVGContainer.innerHTML = '<svg>' + html + '</svg>';
+      var newNodes = reusableSVGContainer.firstChild.childNodes;
+      for (var i = 0; i < newNodes.length; i++) {
+        node.appendChild(newNodes[i]);
+      }
```

Comment:
As @jp7837 pointed out in https://github.com/facebook/react/issues/7358#issuecomment-243289105 - this is going to get every other child - childNodes is a live NodeList. This should be a while loop. I can confirm this is what is happening in IE11.